lib/auth/auth_with_roles_test.go

```diff
@@ -4530,6 +4531,11 @@ func TestGetHeadlessAuthentication(t *testing.T) {
 	username := "teleport-user"
 	otherUsername := "other-user"
 
+	assertTimeout := func(t require.TestingT, err error, i ...interface{}) {
+		require.Error(t, err)
+		require.ErrorContains(t, err, context.DeadlineExceeded.Error(), "expected context deadline error but got: %v", err)
```

Comment:
```suggestion
		require.ErrorIs(t, err, context.DeadlineExceeded)
```

Any reason not to use `ErrorIs` (here and below)?

Reply:
The error is malformed due to the gRPC layer:

```
Error:      	Target error should be in err chain:
            	           	expected: "context deadline exceeded"
            	           	in chain: "rpc error: code = DeadlineExceeded desc = context deadline exceeded"
            	           		"rpc error: code = DeadlineExceeded desc = context deadline exceeded"
            	Test:       	TestUpdateHeadlessAuthenticationState/NOK_admin_approved
            	Messages:   	expected context deadline error but got: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

Reply:
Ok, we should probably update `trail.FromGRPC` to handle `codes.DeadlineExceeded` at some point.